app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+import os
+
+
+class ReadAndDelete:
+    def __init__(self, file_name: str) -> None:
+        self.filename = file_name
+
+    def __enter__(self):
+        self.file = open(self.filename, "r")
+        return self.file
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if os.path.exists(self.filename):
+            os.remove(self.filename)
+
+
+def create_directories(path: str) -> str:
+    directories = path.split("/")
+    full_dir = ""
+    for i in range(len(directories) - 1):
+        full_dir += directories[i] + "/"
+        if not os.path.exists(full_dir):
+            os.mkdir(full_dir)
+    full_dir += directories[len(directories) - 1]
```

Comment:
You can use simpler index - to pick the last value of the array you can try [-1]